Flaky test

```
--- FAIL: Test_chanFilter_Channels (0.00s)
    --- FAIL: Test_chanFilter_Channels/cancelled_context (0.00s)
        processors_test.go:630: 
            	Error Trace:	/home/runner/work/slackdump/slackdump/internal/chunk/control/processors_test.go:630
            	Error:      	Not equal: 
            	            	expected: []structures.EntityItem{}
            	            	actual  : []structures.EntityItem{structures.EntityItem{Id:"C11111111", Oldest:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Latest:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Include:true}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,16 @@
            	            	-([]structures.EntityItem) {
            	            	+([]structures.EntityItem) (len=1) {
            	            	+ (structures.EntityItem) {
            	            	+  Id: (string) (len=9) "C11111111",
            	            	+  Oldest: (time.Time) {
            	            	+   wall: (uint64) 0,
            	            	+   ext: (int64) 0,
            	            	+   loc: (*time.Location)(<nil>)
            	            	+  },
            	            	+  Latest: (time.Time) {
            	            	+   wall: (uint64) 0,
            	            	+   ext: (int64) 0,
            	            	+   loc: (*time.Location)(<nil>)
            	            	+  },
            	            	+  Include: (bool) true
            	            	+ }
            	            	 }
            	Test:       	Test_chanFilter_Channels/cancelled_context
````

--- a/cmd/slackdump/internal/diag/v1.go
+++ b/cmd/slackdump/internal/diag/v1.go
@@ -37,8 +37,19 @@ viewer and other commands.`,
 	Run:         runV1,
 }
 
+var v1Flags = struct {
+	ignoreCopyErrors bool
+}{
+	ignoreCopyErrors: true,
+}
+
+func init() {
+	cmdConvertV1.Flag.BoolVar(&v1Flags.ignoreCopyErrors, "i", v1Flags.ignoreCopyErrors, "ignore copy errors")
+}
+
 func runV1(ctx context.Context, cmd *base.Command, args []string) error {
 	if len(args) != 1 {
+		base.SetExitStatus(base.SInvalidParameters)
 		return fmt.Errorf("must provide a single path to v1.0.x dump")
 	}
 	path := args[0]
@@ -46,10 +57,12 @@ func runV1(ctx context.Context, cmd *base.Command, args []string) error {
 	output := cfg.StripZipExt(cfg.Output)
 
 	if err := os.MkdirAll(output, 0o755); err != nil {
+		base.SetExitStatus(base.SApplicationError)
 		return fmt.Errorf("failed to create output directory: %w", err)
 	}
 	wconn, si, err := bootstrap.Database(output, "v1")
 	if err != nil {
+		base.SetExitStatus(base.SApplicationError)
 		return fmt.Errorf("failed to open database: %w", err)
 	}
 	defer wconn.Close()
@@ -71,11 +84,12 @@ func runV1(ctx context.Context, cmd *base.Command, args []string) error {
 			return fmt.Errorf("failed to create new session: %w", err)
 		}
 		defer erc.Close()
-		if err := convertFile(ctx, erc, filepath.Join(path, p), output); err != nil {
+		if err := convertFile(ctx, erc, filepath.Join(path, p), output, v1Flags.ignoreCopyErrors); err != nil {
 			return fmt.Errorf("failed to convert file %q: %w", p, err)
 		}
 		return nil
 	}); err != nil {
+		base.SetExitStatus(base.SApplicationError)
 		return err
 	}
 
@@ -83,13 +97,13 @@ func runV1(ctx context.Context, cmd *base.Command, args []string) error {
 	return nil
 }
 
-func convertFile(ctx context.Context, erc chunk.Encoder, path string, outputDir string) error {
+func convertFile(ctx context.Context, erc chunk.Encoder, path string, outputDir string, ignoreCopyErr bool) error {
 	src, err := sdv1.NewSource(path)
 	if err != nil {
 		return fmt.Errorf("failed to create source: %w", err)
 	}
 	fsa := fsadapter.NewDirectory(outputDir)
-	conv := convert.NewSourceEncoder(src, fsa, erc, convert.WithIncludeFiles(cfg.WithFiles), convert.WithLogger(cfg.Log), convert.WithIncludeAvatars(false), convert.WithTrgFileLoc(source.MattermostFilepath))
+	conv := convert.NewSourceEncoder(src, fsa, erc, convert.WithIncludeFiles(cfg.WithFiles), convert.WithLogger(cfg.Log), convert.WithIncludeAvatars(false), convert.WithTrgFileLoc(source.MattermostFilepath), convert.WithIgnoreCopyErrors(ignoreCopyErr))
 	if err := conv.Convert(ctx); err != nil {
 		return fmt.Errorf("failed to convert source: %w", err)
 	}

--- a/internal/chunk/control/processors.go
+++ b/internal/chunk/control/processors.go
@@ -119,6 +119,11 @@ var _ processor.Channels = (*chanFilter)(nil)
 // channel matches the filter, and is not excluded or duplicate, it sends the
 // channel ID (as an EntityItem) to the links channel.
 func (c *chanFilter) Channels(ctx context.Context, ch []slack.Channel) error {
+	select {
+	case <-ctx.Done():
+		return context.Cause(ctx)
+	default:
+	}
 LOOP:
 	for _, ch := range ch {
 		if c.memberOnly && (ch.ID[0] == 'C' && !ch.IsMember) {

--- a/internal/convert/convert.go
+++ b/internal/convert/convert.go
@@ -34,6 +34,8 @@ type options struct {
 	includeFiles bool
 	// includeAvatars is a flag to include avatars in the export
 	includeAvatars bool
+	// ignoreCopyErrors is a flag to ignore copy errors
+	ignoreCopyErrors bool
 	// trgFileLoc should return the file location within the target directory
 	trgFileLoc func(*slack.Channel, *slack.File) string
 	// avtrFileLoc should return the avatar file location.
@@ -53,6 +55,12 @@ func WithIncludeFiles(b bool) Option {
 func WithIncludeAvatars(b bool) Option {
 	return func(c *options) {
 		c.includeAvatars = b
+	}
+}
+
+func WithIgnoreCopyErrors(b bool) Option {
+	return func(c *options) {
+		c.ignoreCopyErrors = b
 	}
 }
 

--- a/internal/convert/filecopy.go
+++ b/internal/convert/filecopy.go
@@ -100,8 +100,9 @@ func copy2trg(trgfs fsadapter.FS, trgpath string, srcfs fs.FS, srcpath string) e
 }
 
 type avatarcopywrapper struct {
-	fsa  fsadapter.FS
-	avst source.Storage
+	fsa          fsadapter.FS
+	avst         source.Storage
+	ignoreErrors bool
 }
 
 func (a *avatarcopywrapper) Users(ctx context.Context, users []slack.User) error {
@@ -135,6 +136,10 @@ func (a *avatarcopywrapper) copyAvatar(u slack.User) error {
 	dstloc := path.Join(chunk.AvatarsDir, path.Join(source.AvatarParams(&u)))
 	src, err := fsys.Open(srcloc)
 	if err != nil {
+		if a.ignoreErrors {
+			slog.Warn("copying avatar", "user", u.ID, "error", err)
+			return nil
+		}
 		return err
 	}
 	defer src.Close()
@@ -152,11 +157,19 @@ func (a *avatarcopywrapper) copyAvatar(u slack.User) error {
 }
 
 type filecopywrapper struct {
-	fc copier
+	fc           copier
+	ignoreErrors bool
 }
 
-func (f *filecopywrapper) Files(_ context.Context, ch *slack.Channel, parent slack.Message, _ []slack.File) error {
-	return f.fc.Copy(ch, &parent)
+func (f *filecopywrapper) Files(ctx context.Context, ch *slack.Channel, parent slack.Message, _ []slack.File) error {
+	if err := f.fc.Copy(ch, &parent); err != nil {
+		if f.ignoreErrors {
+			slog.WarnContext(ctx, "copying file", "error", err)
+			return nil
+		}
+		return err
+	}
+	return nil
 }
 
 func (f *filecopywrapper) Close() error {

--- a/internal/convert/src2enc.go
+++ b/internal/convert/src2enc.go
@@ -64,7 +64,10 @@ func (s *SourceEncoder) Convert(ctx context.Context) error {
 	var cp processor.Conversations = rec
 	if s.opts.includeFiles && s.src.Files().Type() != source.STnone {
 		fc := NewFileCopier(s.src, s.fsa, source.MattermostFilepath, s.opts.includeFiles)
-		cp = processor.PrependFiler(rec, &filecopywrapper{fc})
+		cp = processor.PrependFiler(rec, &filecopywrapper{
+			fc:           fc,
+			ignoreErrors: s.opts.ignoreCopyErrors,
+		})
 	}
 	channels, err := s.src.Channels(ctx)
 	if err != nil {

--- a/internal/viewer/templates/styles.html
+++ b/internal/viewer/templates/styles.html
@@ -57,7 +57,8 @@
         padding-bottom: 1.5rem;
         flex: none;
         background-color: var(--sidebar-bg);
-        color: var(--text-color)
+        color: var(--text-color);
+	overflow-y: auto;
     }
 
     /* Sidebar channel heading */

--- a/source/doc.go
+++ b/source/doc.go
@@ -1,0 +1,64 @@
+// Package source provides archive readers for different output formats.
+//
+// Currently, the following formats are supported:
+//   - archive
+//   - database
+//   - dump
+//   - Slack Export
+//
+// One should use [Load] function to load the source from the file system.  It
+// will automatically detect the format and return the appropriate reader.
+//
+// All sources implement the [Sourcer] interface, which provides methods to
+// retrieve data from the source.  The [Resumer] interface is implemented by
+// sources that can be resumed.  The [SourceResumeCloser] interface is
+// implemented by sources that can be resumed and closed.
+//
+// There are two distinct error types of interest, [ErrNotFound] and
+// [ErrNotSupported].  See their documentation for more information.
+//
+// # What is a Source?
+//
+// A source is a generic interface over the data source.  It can be implemented
+// not only around Slack data, but also around any other messenger data, as it
+// represents common entities every messenger has, for example "message",
+// "thread", "channel", for Telegram would be "message", "reply", "chat"
+// respectively.  The caveat is that non-Slack source's entities would need to
+// be converted to Slack entities, which shouldn't be hard to do, unless you're
+// aiming to replicate formatting.
+//
+// In this package, for now, the source is implemented only for data
+// originating from Slackdump and Slack.
+//
+// # Loading a Source
+//
+// If you know what source you are loading, or need  a particular source type
+// you can use a concrete Open* function, such as [OpenDatabase].  If you
+// don't, you can use [Load] function that will determine the source type and
+// return the appropriate source.
+//
+// It is a good idea to defer the closing of the source, as it may be a
+// database connection or a file handle.  The [SourceResumeCloser] interface,
+// returned by [Load] implements [io.Closer] interface, so you can use it with
+// defer statement.  For example:
+//
+//	src, err := source.Load(ctx, "path/to/source")
+//	if err != nil {
+//	    log.Fatal(err)
+//	}
+//	defer src.Close()
+//
+// Within your code, you can call the Type method which will return the source
+// type.
+//
+// # Source Types
+//
+// The source type returned by [Type] is a bitmask, you can use [Has] method to
+// check if particular flag is set.  Flag constants all start with F* and have
+// [Flags] type.
+//
+// The source type returned by Type method on a particular source will only
+// have the type flag set, without any additional flags (as of v3.1.0, but this
+// may change in future versions, so always use Has method to be on the safe
+// side).
+package source

--- a/source/source.go
+++ b/source/source.go
@@ -1,21 +1,3 @@
-// Package source provides archive readers for different output formats.
-//
-// Currently, the following formats are supported:
-//   - archive
-//   - database
-//   - dump
-//   - Slack Export
-//
-// One should use [Load] function to load the source from the file system.  It
-// will automatically detect the format and return the appropriate reader.
-//
-// All sources implement the [Sourcer] interface, which provides methods to
-// retrieve data from the source.  The [Resumer] interface is implemented by
-// sources that can be resumed.  The [SourceResumeCloser] interface is
-// implemented by sources that can be resumed and closed.
-//
-// There are two distinct error types of interest, [ErrNotFound] and
-// [ErrNotSupported].  See their documentation for more information.
 package source
 
 import (
@@ -106,14 +88,22 @@ type Flags int8
 
 const (
 	// container
+	// FDirectory indicates that the source is a directory.
 	FDirectory Flags = 1 << iota
+	// FZip indicates that the source is a ZIP archive.
 	FZip
 	// main content
+	// FChunk is set on directories with json.gz files.
 	FChunk
+	// FExport is set on Slack exoprt directories or ZIP files.
 	FExport
+	// FDump is set on a Slackdump Dump format directory or ZIP file.
 	FDump
+	// FDatabase indicates that the source is a SQLite database.
 	FDatabase
 
+	// FUnknown is returned if the source type is not supported or
+	// can not be determined.
 	FUnknown Flags = 0
 )
 

